### ISSUE TITLE
How to config WandbLogger with LightingCLI ?

### ISSUE THREAD

**Issue by real-junjiezhang** [Lightning-AI/pytorch-lightning#20064]
### Bug description

I want use wandb to track the change of gradients during training. However I don't know how to write this with LightingCLI. Can someone show me a template regarding how to use wandb to track gradients in config.yaml? Any support will be highly appreciated
Thanks in advance.

### What version are you seeing the problem on?

master

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- PyTorch Lightning Version (e.g., 1.5.0):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
```

</details>

**Comment by awaelchli** [comment]
I think you should be able to configure any object (be it a logger, or callback etc.) in the config file for LightningCLI/jsonargparse like so:

```
trainer:
  logger:
    - class_path: lightning.pytorch.loggers.wandb.WandbLogger
      init_args:
        project: test-project
        # put your wandb arguments here
```

([from CLI docs with callbacks](https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_advanced_3.html#trainer-callbacks-and-arguments-with-class-type))

To log the gradient norm, use the logging facilities in the LightningModule:
https://lightning.ai/docs/pytorch/stable/debug/debugging_intermediate.html#look-out-for-exploding-gradients

**Comment by SkafteNicki** [comment]
`wandb` comes build in with tracking of gradients through the `.watch` method, so calling that will enable gradient logging. However, it is not possible to call this method directly through lightning cli. Instead what is begin asked for can be done with a custom callback. Example script + config:
```python
from lightning.pytorch.demos import BoringDataModule, BoringModel
from lightning.pytorch.cli import LightningCLI
from lightning.pytorch.loggers import WandbLogger
from lightning.pytorch.callbacks import Callback
from typing import Optional

class WatchModel(Callback):
    def __init__(self, log_type: Optional[str] = "gradients", log_graph: bool = True):
        self.log_type = log_type
        self.log_graph = log_graph

    def on_fit_start(self, trainer, pl_module):
        logger = trainer.logger
        if isinstance(logger, WandbLogger):
            logger.watch(pl_module, log=self.log_type, log_freq=trainer.log_every_n_steps, log_graph=self.log_graph)


if __name__ == "__main__":
    cli = LightningCLI(
        BoringModel,
        BoringDataModule,
        save_config_callback=None,
    )
```
```yaml
trainer:
  callbacks:
    - class_path: __main__.WatchModel
      init_args:
        log_type: "gradients"
        log_graph: true
  logger:
    - class_path: lightning.pytorch.loggers.WandbLogger
      init_args:
        project: "test-project"
  log_every_n_steps: 50
  max_epochs: 3
```
Closing issue.
